docs/building.md

```diff
@@ -0,0 +1,176 @@
+# Cloning and building
+
+Fobnail project consists of 3 main repositories with code:
+
+- [Fobnail firmware](https://github.com/fobnail/fobnail) (Rust)
+- [Platform Owner](https://github.com/fobnail/fobnail-platform-owner) (C)
+- [Attester](https://github.com/fobnail/fobnail-attester) (C)
+
+There is also [Fobnail SDK](https://github.com/fobnail/fobnail-sdk) that
+simplifies building of firmware by using Docker container, but only
+[one script](https://github.com/fobnail/fobnail-sdk/blob/main/run-fobnail-sdk.sh)
+from this repository is actually needed.
+
+Following steps were successfully performed on Ubuntu 22.04.1 LTS.
+
+## Fobnail Token firmware
+
+Start by installing prerequisites and adding current user to `docker` group:
+
+```shell
+sudo apt -y install git docker.io
+sudo usermod -aG docker $USER
+```
+
+Log out of the desktop session then log in again - this is required after making
+the change to user groups. You may confirm it by running `groups`.
+
+Clone the repository along with its submodules:
+
+```shell
+git clone https://github.com/fobnail/fobnail --recurse-submodules
+```
+
+Build process for Token firmware is by far the most complicated one. For that
+reason Fobnail SDK is strongly suggested. To use it, simply download script
+`run-fobnail-sdk.sh` either by cloning whole repository or just this one script
+and save it to a directory in your `PATH` environment variable:
+
+```shell
+sudo wget -O /usr/local/bin/run-fobnail-sdk.sh \
+  https://raw.githubusercontent.com/fobnail/fobnail-sdk/main/run-fobnail-sdk.sh
+sudo chmod +x /usr/local/bin/run-fobnail-sdk.sh
+```
+
+With the script installed, one can prepare for the build process itself.
+Firmware can be build for a physical Fobnail Token or simulated on PC, with
+different preparatory steps required. In both cases, first execution of
+`build.sh` downloads Docker container image, which can be both time- and
+bandwidth-consuming.
+
+#### Environment variables common for both targets
+
+Fobnail Token firmware is configured with environment variables passed to
+`build.sh`.
+
+Note that due to the way Docker mounts directories all files and directories
+pointed to by following variables must be located somewhere in `fobnail`
+directory.
+
+- `FOBNAIL_PO_ROOT` - **required** option, must point to valid PEM or DER file
+  with Platform Owner's root certificate. See [this document](/keys_and_certificates/#platform-owner-certificate-chain)
```

Comment:
I'm trying to get through PO document. Commands like this:

```
openssl req -newkey rsa:2048 -nodes -keyout $ROOT_PRIV -x509 -days 365 \
        -out $ROOT_CERT -config $ROOT_CONFIG
```

Are confusing, despite all enthronement variables are explained below the command it is not clear if it matters to developer what those values would be.  IMO we over-complicate the process. Also order is reversed. First we show command which e.g. needs config, then we show how to do config. I think this should be changed. I will try propose some modification.

Reply:
Another problem in description is using different names. We should be consistent to avoid confusing of potential users.

Reply:
Added some improvement suggestions here: https://github.com/fobnail/docs/pull/49
Please let me know WDYT.